File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/JawWorm_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class JawWorm_Model extends MonsterInfo{
    public static Id = "JawWorm"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(42,46);
        else
            return Mathf.RandomRange(40,44);
    }
    
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/jawWorm/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/jawWorm/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/jawWorm/skeleton.png"));
    }

    get Name(): string {
        return "大颚虫";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/JawWorm_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class JawWorm_Model : MonsterInfo
    {
        public static string Id = "JawWorm";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(42,46) : Random.Range(40,44);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/jawWorm/skeleton.atlas",
                    "Monsters/theBottom/jawWorm/skeleton.json", "Monsters/theBottom/jawWorm/skeleton.png");
            }
        }

        public override string Name => "大颚虫";

        public JawWorm_Model(Vector2 pos) : base(pos)
        {
        }
    }
}